crates/ln-dlc-node/src/dlc_custom_signer.rs

```diff
@@ -292,11 +297,25 @@ impl KeysInterface for CustomKeysManager {
     }
 
     fn get_destination_script(&self) -> Script {
-        self.keys_manager.get_destination_script()
+        let address = self
+            .wallet
+            .get_new_address()
+            .expect("Failed to retrieve new address from wallet.");
+        address.script_pubkey()
     }
 
     fn get_shutdown_scriptpubkey(&self) -> ShutdownScript {
-        self.keys_manager.get_shutdown_scriptpubkey()
+        let address = self
+            .wallet
+            .get_new_address()
```

Comment:
🤔 should we use last unused address here?

Reply:
Yes please, let's use `last_unused` here. Otherwise we get gaps in the derivation paths.

Reply:
It's getting a `last_unused` under the hood :D